File: app/src/main/java/com/pnuema/bible/ui/viewholders/VersionSelectionViewHolder.java
package com.pnuema.bible.ui.viewholders;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.pnuema.bible.R;
import com.pnuema.bible.data.IVersion;

public class VersionSelectionViewHolder extends RecyclerView.ViewHolder {
    private final TextView mContentView;
    private IVersion mItem;

    public VersionSelectionViewHolder(View view) {
        super(view);
        mContentView = view.findViewById(R.id.version);
    }

    @Override
    public String toString() {
        return super.toString() + " '" + mContentView.getText() + "'";
    }

    @Nullable
    public TextView getContentView() {
        return mContentView;
    }

    @Nullable
    public IVersion getItem() {
        return mItem;
    }

    public void setItem(@NonNull IVersion mItem) {
        this.mItem = mItem;
    }
}

File: app/src/main/java/com/pnuema/bible/android/ui/viewholders/VersionSelectionViewHolder.kt
package com.pnuema.bible.android.ui.viewholders

import android.graphics.Typeface
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.pnuema.bible.android.R
import com.pnuema.bible.android.data.IVersion
import com.pnuema.bible.android.databinding.ListitemVersionBinding
import com.pnuema.bible.android.statics.CurrentSelected
import com.pnuema.bible.android.ui.dialogs.VersionSelectionListener

class VersionSelectionViewHolder(
        parent: ViewGroup,
        private val listener: VersionSelectionListener,
        private val binding: ListitemVersionBinding = ListitemVersionBinding.inflate(LayoutInflater.from(parent.context), parent, false)
) : RecyclerView.ViewHolder(binding.root) {
    override fun toString(): String {
        return super.toString() + " '" + (itemView as TextView).text + "'"
    }

    fun bind(version: IVersion) {
        binding.root.text = version.getDisplayText()

        val currentIsSelected = !TextUtils.isEmpty(CurrentSelected.version) && CurrentSelected.version == version.abbreviation

        binding.root.setTextColor(ContextCompat.getColor(binding.root.context, if (currentIsSelected) R.color.primary else R.color.primary_text))
        binding.root.setTypeface(null, if (currentIsSelected) Typeface.BOLD else Typeface.NORMAL)

        binding.root.setOnClickListener {
            listener.onVersionSelected(version.abbreviation)
        }
    }
}